packages/shared/src/retry.ts

```diff
@@ -0,0 +1,108 @@
+type Milliseconds = number;
+
+type RetryOptions = Partial<{
+  /**
+   * The initial delay before the first retry.
+   * @default 125
+   */
+  firstDelay: Milliseconds;
```

Comment:
How about `initialDelay`? How will it be different from the "immediate" delay?

Reply:
Sounds good, will update the code accordingly 👍🏻